frontend/src/components/HelpPopup/HelpPopup.test.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+// import userEvent from "@testing-library/user-event";
+
+import HelpPopup from "./HelpPopup";
+import userEvent from "@testing-library/user-event";
+
+jest.mock(
+  "./HelpPopup",
+  () =>
+    function () {
+      return <span>HelpPopup</span>;
+    },
+);
+
+test("TID 3.1. Renders HelpPopup component", () => {
+  const dom = render(<HelpPopup />);
+
+  // dom.getByText("? Help");
+
+  // clicking the popup expand button
+  const button = dom.container.querySelector("popupButton");
+
+  if (button === null) {
+    throw new Error("Button didnt render");
+  } else {
+    userEvent.click(button);
+  }
+});
```

Comment:
Once we remove the mocking module we can now test our HelpComponent as if it were rendered in an actual browser.

I suggest as an approach to testing are

Checking if File Upload, Processed Files and Predicted Files are rendered along with ? Help.

Also testing what gets rendered when `? Help` gets clicked and what happens if the buttons for the paragraph gets clicked.

For example you can do this to test collapsing and expanding of your paragraphs

```
// Assuming ? help has already been clicked
expect(getByTestId("Paragraph 0").getAttribute("values")).toEqual("closed"); // We are checking if Paragraph 0 is closed
userEvent.click(getByTestId("button 0"));  // Expand the paragraph
expect(getByTestId("Paragraph 0").getAttribute("values")).toEqual("opened"); // It should now be open
// Collapse File Upload again
userEvent.click(getByTestId("button 0"));
expect(getByTestId("Paragraph 0").getAttribute("values")).toEqual("closed");
```